### PR TITLE
kubernetes_server_version data source

### PR DESCRIPTION
**Pull request by daemon4d-us**
### Acceptance tests
- [ Yes] Have you added an acceptance test for the functionality being added?
- [ Yes] Have you run the acceptance tests on this branch?

Output from acceptance testing:

```
dsidorenko@dsidorenko01-la terraform-provider-kubernetes % make testacc TESTARGS='-run=TestAccKubernetesDataSourceServerVersion_basic'
==> Checking that code complies with gofmt requirements...
go vet .
TF_ACC=1 go test "/Users/dsidorenko/go/src/github.com/hashicorp/terraform-provider-kubernetes/kubernetes" -v -vet=off -run=TestAccKubernetesDataSourceServerVersion_basic -parallel 8 -timeout 3h
=== RUN   TestAccKubernetesDataSourceServerVersion_basic
=== PAUSE TestAccKubernetesDataSourceServerVersion_basic
=== CONT  TestAccKubernetesDataSourceServerVersion_basic
--- PASS: TestAccKubernetesDataSourceServerVersion_basic (3.64s)
PASS
ok      github.com/hashicorp/terraform-provider-kubernetes/kubernetes   4.069s
```

### Release Note
Release note for [CHANGELOG](https://github.com/hashicorp/terraform-provider-kubernetes/blob/main/CHANGELOG.md):

```release-note
data kubernetes_server_version added. 
...
```

### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### CODE REVIEW

File: kubernetes/data_source_kubernetes_server_version.go
Comment (torqu3e) [inline kubernetes/data_source_kubernetes_server_version.go:38]: Was gonna open a PR and saw yours. Mine had extra version information to match the `kubectl version` command that can be added here.

```
	server_version := map[string]string{
		"buildDate":    sv.BuildDate,
		"compiler":     sv.Compiler,
		"gitCommit":    sv.GitCommit,
		"gitTreeState": sv.GitTreeState,
		"gitVersion":   sv.GitVersion,
		"goVersion":    sv.GoVersion,
		"major":        sv.Major,
		"minor":        sv.Minor,
		"platform":     sv.Platform,
		"version":      sv.String(),
	}
	d.Set("server_version", server_version)
```

Comment (daemon4d-us) [reply kubernetes/data_source_kubernetes_server_version.go:38]: This is a good one. I thought about this as well. Let me add this to my PR if you don't mind.

**Comment by daemon4d-us** [comment]
Hi @torqu3e Does this look better?

**Comment by alexsomesan** [comment]
@daemon4d-us 
Changes generally look good and this is a useful feature to have in the provider.
In order for community PRs to be merged contributors need to agree to the CLA. Please see the prompt above from hashicorp-cla bot.

**Comment by daemon4d-us** [comment]
Hm, something is not working quite good here for me. I have actually signed the CLA, but still appears in Pending state here.

**Comment by daemon4d-us** [comment]
@arybolovlev Thanks for figuring this out. The CLA check is green now. But since I've changed history the PR needs one more approval.

**Comment by daemon4d-us** [comment]
Anyone can take a look one more time?

**Comment by BBBmau** [comment]
Please refer to @arybolovlev's comments, still missing documentation and changelog entry in order for this PR to get merged.

**Comment by BBBmau** [comment]
any update on this? @daemon4d-us

**Review by BBBmau — APPROVED** [review]
Approved after some manual tests as well as adding in documentation with the help of tfplugindocs, thanks again for opening this! @daemon4d-us

**Pull request merged**
